### DISCUSSION TITLE
Reduce the size of the library. [pyinstaller]

### DISCUSSION THREAD

**Discussion by yeus** [pydantic/pydantic#3922] [Question]
I am using pyinstaller to package a software that I am working on for local install. I am generally not thrilled by the size of the package. The largest chunk in it is pytorch (roughly 400MiB)...  I already did quiet bit of optimization on it and reduced the total size to approx. 1GiB. What surprised me though is that the second largest chunk in it is produced by pydantic with 89MiB!

Here is a breakdown of my project: The green block is the one from pydantic after packaging with pyinstaller...

![image](https://user-images.githubusercontent.com/736695/159259729-80de5771-8217-4868-a372-22592f95aa2e.png)

and here is another breakdown directly from the virtual env/site_packages directory which shows that here as well pydantic is on par with sklearn regarding chunksize:

![image](https://user-images.githubusercontent.com/736695/159260806-11eb4ee9-be12-425b-a4ad-3e71b7941a19.png)

Is there anyway to reduce this chunk? I am using fastapi in it.. so pydantic is a necessary requirement. Why is the size of the library so huge?  89MiB seems quiet crazy to me for some validation.. larger than sklearn, pandas and all these other libraries...

I read here:  https://github.com/samuelcolvin/pydantic/issues/2276 that it might be related to cython? --> can I simply remove the copiled *.so files and it would still work? Would it significantly slow down pydantic or what is the main disadvantage? -->In my software the slowdown wouldnt be as important...

**Comment by samuelcolvin** [comment]
> can I simply remove the copiled *.so files and it would still work?

should do, best to try it.

> Would it significantly slow down pydantic or what is the main disadvantage?

Roughly 50% slow down, but it varies by application, again, best to try.
